source/handbook/contributing/coding_standards.rst

```diff
@@ -0,0 +1,158 @@
+    * file that was distributed with this source code.
+    */
+
+    namespace ONGR\FooBundle;
+
+    use ONGR\WhateverBundle\WhateverUtility;
+
+    namespace Acme;
+
+    /**
+    * Coding standards demonstration.
+    */
+    class FooBar
+    {
+        const SOME_CONST = 42;
+
+        /**
+         * @var FooService
+         */
+        private $fooBar;
+
+        /**
+         * @param string $dummy Some argument description.
+         */
+        public function __construct($dummy)
+        {
+            $this->fooBar = $this->transformText($dummy);
+        }
+
+        /**
+         * Transform text in a magic way.
+         *
+         * @param string $dummy   Some argument description.
+         * @param array  $options Optional options.
+         *
+         * @return string|null Transformed input.
+         * @throws \RuntimeException
+         */
+        private function transformText($dummy, array $options = [])
+        {
+            $mergedOptions = array_merge(
+                [
+                    'some_default' => 'values',
+                    'another_default' => 'more values',
+                ],
+                $options
+            );
+
+            if (true === $dummy) {
+                return;
+            }
+
+            if ('string' === $dummy) {
+                if ('values' === $mergedOptions['some_default']) {
+                    return substr($dummy, 0, 5);
+                }
+
+                return ucwords($dummy);
+            }
+
+            throw new \RuntimeException(sprintf('Unrecognized dummy option "%s"', $dummy));
+        }
+    }
+
+..
+
+   Note: Don't forget to leave single empty line before license header, namespace, uses and class comment block. All uses should be listed in alphabetical order without empty lines.
+
+Tickets and releases
+--------------------
+
+Task has been completed, if:
+
+1. Feature is implemented.
+2. New functionality is covered with automated tests.
+3. Feature has been documented in documentation repository.
+
+Release:
+
+1. Every release must have some valuable description or list of changes (links to wiki are recommended).
+2. Every commit message should contain short description of what was done in it.
+
+   i. No need to include feature suffix. PR's are used for grouping commits into features.
+   ii. It's not recommended to mention issue number (e.g. Fixed price handling, closes #123). Better to link commit hash in the issue. Otherwise, it's hard to change wrong issue number in the commit.
+
+Documenting code
+----------------
+
+1. Use ``{@inheritDoc}`` when extending abstract methods or implementing interfaces instead of rewriting anything.
+2. If method does not return any result, ``@return`` annotation must be omitted.
+3. Comments must (1) start with capital letter, (2) have a single space between comment symbols and first letter and (3) must NOT include period at the end of the comment if it is one sentence. E. g. ``// This is a short comment``
```

Comment:
Inline comments must end in full-stops, exclamation marks, or question marks <- exact error message, when using this example :)